layers/+lang/flow-type/funcs.el

```diff
@@ -0,0 +1,133 @@
+(defun flow-type/--no-auto-start-arg ()
+  (if flow-type-no-auto-start "--no-auto-start" ""))
+
+(defun flow-type/call-process-on-buffer-to-string (command)
+  (with-output-to-string
+    (call-process-region (point-min) (point-max) shell-file-name nil standard-output nil shell-command-switch command)))
+
+(defun flow-type/get-info-from-def (info)
+  (let ((type (alist-get 'type info)))
+    (if (not (string-equal type "(unknown)"))
+        type)))
+
+(defun flow-type/get-reasons-from-def (info)
+  (let ((reasons (alist-get 'reasons info)))
+    (if (> (length reasons) 0)
+        (alist-get 'desc (aref reasons 0)))))
+
+
+(defun flow-type/colorize-type (text)
+  ;; if web-mode is installed, use that to syntax highlight the type information
+  (if (configuration-layer/package-usedp 'web-mode)
+      (with-temp-buffer
+        (web-mode)
+        (insert text)
+        (web-mode-set-content-type "jsx")
+        (buffer-string))
+    ;; otherwise, just return the plain text
+    text))
+
+(defun flow-type/describe-info-object (obj)
+  (let ((err (alist-get 'error obj)))
+    (if err (error err)))
+  (let ((info (flow-type/get-info-from-def obj)))
+    (if info
+        (flow-type/colorize-type info)
+      ;; even if the 'info' field is unknown, there's sometimes useful information in the 'reasons' structure.
+      (flow-type/get-reasons-from-def obj))))
+
+(defun flow-type/flow-binary ()
+    (or
+     ;; Try to find flow in node_modules (via 'npm install flow-bin')
+      (let ((root (locate-dominating-file buffer-file-name "node_modules")))
```

Comment:
Instead of duplicating this logic in multiple(?) layers maybe consider `add-node-modules-path` as in  https://github.com/syl20bnr/spacemacs/pull/8621